spec/support/guard_platform.rb

```diff
@@ -0,0 +1,83 @@
+class PlatformGuard
```

Comment:
I guess the code has been taken from MSpec - https://github.com/ruby/mspec/blob/master/lib/mspec/guards/platform.rb. Did you make any changes to make it working? If so could you add a comment for each patch please?

How many changes did you make? Is it possible to require original implementation here and monkey-patch it on the fly?

Also I've noticed that in MSpec it inherits from the `SpecGuard`, but here it's a subclass of `Object`. Is there any reason for doing that?

Reply:
You're correct it's a simplified version of this file.

> Did you make any changes to make it working?

Yes, I've removed unused methods and I'm using environment variable instead of `RbConfig`.

>  If so could you add a comment for each patch please?

Sure

> How many changes did you make? Is it possible to require original implementation here and monkey-patch it on the fly?

Maybe but I believe some parts are "static", for instance the following code will be executed right away ?

```rb
  POINTER_SIZE = begin
    require 'rbconfig/sizeof'
    RbConfig::SIZEOF["void*"] * 8
  rescue LoadError
    WORD_SIZE
  end
```

> Also I've noticed that in MSpec it inherits from the SpecGuard, but here it's a subclass of Object. Is there any reason for doing that?

`SpecGuard` was used in the orignal implementation but it's not needed here.
In the original implementation, `run_if` and `run_unless` are called but in this simplified implementation I just do everything in this file.

Reply:
> the following code will be executed right away

Yes, but I think we can handle `LoadError` like MRI would do and fallback to `WORD_SIZE` (that's what you do manually, right?)

Reply:
Yes, I will try to remove as much code as I can :+1:

Reply:
It seems to be working if I just redefine `HOST_OS` and `POINTER_SIZE`